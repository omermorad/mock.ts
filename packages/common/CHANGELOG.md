## [2.0.3](https://github.com/omermorad/mockingbird/compare/@mockingbird/common@2.0.3...@mockingbird/common@2.0.3) (2021-11-17)


### Features

* **common:** add regex constructor to exact value ([10263fb](https://github.com/omermorad/mockingbird/commit/10263fb17287eb86516bd4778960586106011c2f))



## 2.1.1 (2021-08-21)



# 2.1.0 (2021-08-21)



