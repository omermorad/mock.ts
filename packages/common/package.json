{
  "name": "@mockingbird/common",
  "version": "3.0.0-rc.1",
  "license": "MIT",
  "description": "Mockingbird Common Package",
  "types": "dist/index.d.ts",
  "contributors": [
    {
      "name": "Omer Morad",
      "email": "omer.moradd@gmail.com"
    }
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/omermorad/mockingbird.git"
  },
  "bugs": {
    "url": "https://github.com/omermorad/mockingbird/issues"
  },
  "readme": "https://github.com/omermorad/mockingbird/tree/refactor/master/packages/common/README.md",
  "scripts": {
    "prebuild": "npx rimraf dist",
    "build": "tsc",
    "watch": "tsc --watch",
    "test": "exit 0;",
    "lint": "eslint 'src/**/*.ts'",
    "lint:fix": "eslint 'src/**/*.ts' --fix"
  },
  "files": [
    "dist",
    "README.md",
    "CHANGELOG.md"
  ],
  "dependencies": {
    "@types/faker": "^5.5.7",
    "faker": "^5.5.3"
  },
  "devDependencies": {
    "ts-loader": "^6.2.2",
    "ts-node": "8.10.2",
    "tsconfig-paths": "^3.9.0",
    "typescript": "^3.9.7"
  },
  "publishConfig": {
    "access": "public"
  }
}
