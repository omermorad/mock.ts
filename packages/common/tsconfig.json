{
  "extends": "../tsconfig.build.json",
  "compilerOptions": {
    "baseUrl": "./",
    "rootDir": "./src",
    "outDir": "./dist",
    "target": "es5"
  },
  "exclude": [
    "index.ts",
    "node_modules",
    "test",
    "src/**/*.test.ts"
  ],
  "include": ["src/"]
}