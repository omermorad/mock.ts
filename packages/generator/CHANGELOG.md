# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [2.0.0-alpha.2](https://github.com/omermorad/mockingbird-ts/compare/mockingbird-ts@2.0.0-alpha.1...mockingbird-ts@2.0.0-alpha.2) (2021-07-23)

**Note:** Version bump only for package mockingbird-ts





# [2.0.0-alpha.1](https://github.com/omermorad/mockingbird-ts/compare/mockingbird-ts@2.0.0...mockingbird-ts@2.0.0-alpha.1) (2021-07-23)


### Bug Fixes

* **types:** add build to export a js file to include the faker instance (for runtime) ([#61](https://github.com/omermorad/mockingbird-ts/issues/61)) ([f4e3092](https://github.com/omermorad/mockingbird-ts/commit/f4e3092e683eb9c288d4e879113e71f74ec5038a))


### Features

* **mockingbird-ts:** add mock factory fluent/builder api ([#60](https://github.com/omermorad/mockingbird-ts/issues/60)) ([cc5710d](https://github.com/omermorad/mockingbird-ts/commit/cc5710ded33401cae25782bb8e87efe1355024aa)), closes [#42](https://github.com/omermorad/mockingbird-ts/issues/42)


### Reverts

* **repo:** release packages ([7f9390d](https://github.com/omermorad/mockingbird-ts/commit/7f9390d051f9c9c9c3eb172f4db8a9fe533b03c4))


### BREAKING CHANGES

* **mockingbird-ts:** MockGenerator is not exported anymore, use MockFactory instead





# Change Log

# [2.0.0-alpha.0](https://github.com/omermorad/mockingbird-ts/compare/mockingbird-ts@2.0.0...mockingbird-ts@2.0.0-alpha.0) (2021-07-22)


### Features

* **mockingbird-ts:** add mock factory fluent/builder api ([#60](https://github.com/omermorad/mockingbird-ts/issues/60)) ([cc5710d](https://github.com/omermorad/mockingbird-ts/commit/cc5710ded33401cae25782bb8e87efe1355024aa)), closes [#42](https://github.com/omermorad/mockingbird-ts/issues/42)

### BREAKING CHANGES

* **mockingbird-ts:** `MockFactory` is no longer a class, but a function returning a builder API


# [1.1.0](https://github.com/omermorad/mockingbird-ts/compare/v1.0.0...v1.1.0) (2021-07-08)


### Features

* add new types and sample ([acc1776](https://github.com/omermorad/mockingbird-ts/commit/acc1776b50360fe745983b266b7e9a5da1ee9f4f))
* add typeorm sample, change types ([f801a59](https://github.com/omermorad/mockingbird-ts/commit/f801a59e0811e92f72a9c28069a809bfb9624564))

# [1.1.0-next.1](https://github.com/omermorad/mockingbird-ts/compare/v1.0.0...v1.1.0-next.1) (2021-02-10)


### Features

* add new types and sample ([acc1776](https://github.com/omermorad/mockingbird-ts/commit/acc1776b50360fe745983b266b7e9a5da1ee9f4f))

# 1.0.0 (2021-02-09)


### Bug Fixes

* **signatures:** remove redundant signature for Date type overload ([e4fbbc1](https://github.com/omermorad/mockingbird-ts/commit/e4fbbc18eb710bc181ef7a2d98490132cf4771df))


### Code Refactoring

* **types:** add some new types to be more accurate ([d4eba88](https://github.com/omermorad/mockingbird-ts/commit/d4eba8866b000f3507d3f1a2cf8881d3040972fc))


### Continuous Integration

* **release:** add beta as pre-release branch ([aed0c09](https://github.com/omermorad/mockingbird-ts/commit/aed0c0906f22096f7ecabe9b75aee04d410e5cef))


### Features

* **release:** release new beta version ([bda5aa7](https://github.com/omermorad/mockingbird-ts/commit/bda5aa74b957220e90605881556dacffc538a130))
* alpha release ([#12](https://github.com/omermorad/mockingbird-ts/issues/12)) ([5682bb4](https://github.com/omermorad/mockingbird-ts/commit/5682bb4c21df4d546166c613f8ed7fff937dc3dc))
* faker.ts initial ([#6](https://github.com/omermorad/mockingbird-ts/issues/6)) ([714fc05](https://github.com/omermorad/mockingbird-ts/commit/714fc05d1fdd93e1a709ebe183776dd28d0681bf)), closes [#4](https://github.com/omermorad/mockingbird-ts/issues/4) [#7](https://github.com/omermorad/mockingbird-ts/issues/7) [#8](https://github.com/omermorad/mockingbird-ts/issues/8)


### BREAKING CHANGES

* **types:** New MockedClass type
* **release:** Add 'tslib' as dependency

# 1.0.0-rc.1 (2021-02-05)


### Bug Fixes

* **signatures:** remove redundant signature for Date type overload ([e4fbbc1](https://github.com/omermorad/mockingbird-ts/commit/e4fbbc18eb710bc181ef7a2d98490132cf4771df))


### Continuous Integration

* **release:** add beta as pre-release branch ([aed0c09](https://github.com/omermorad/mockingbird-ts/commit/aed0c0906f22096f7ecabe9b75aee04d410e5cef))


### Features

* **release:** release new beta version ([bda5aa7](https://github.com/omermorad/mockingbird-ts/commit/bda5aa74b957220e90605881556dacffc538a130))
* alpha release ([#12](https://github.com/omermorad/mockingbird-ts/issues/12)) ([5682bb4](https://github.com/omermorad/mockingbird-ts/commit/5682bb4c21df4d546166c613f8ed7fff937dc3dc))
* faker.ts initial ([#6](https://github.com/omermorad/mockingbird-ts/issues/6)) ([714fc05](https://github.com/omermorad/mockingbird-ts/commit/714fc05d1fdd93e1a709ebe183776dd28d0681bf)), closes [#4](https://github.com/omermorad/mockingbird-ts/issues/4) [#7](https://github.com/omermorad/mockingbird-ts/issues/7) [#8](https://github.com/omermorad/mockingbird-ts/issues/8)


### BREAKING CHANGES

* **release:** Add 'tslib' as dependency

# 1.0.0-alpha.1 (2021-01-28)

### Features

* Mockingbird alpha release ([#11](https://github.com/omermorad/mockingbird-ts/issues/11)) ([67511b7](https://github.com/omermorad/faker.ts/commit/67511b7bc7792e06ac54c752b0ac96ee5337fd35)), closes [#4](https://github.com/omermorad/faker.ts/issues/4) [#7](https://github.com/omermorad/faker.ts/issues/7) [#8](https://github.com/omermorad/faker.ts/issues/8)
