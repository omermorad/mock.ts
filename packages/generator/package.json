{
  "name": "mockingbird-ts",
  "version": "2.0.0-alpha.2",
  "license": "MIT",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "description": "Super Simple, Yet Powerful, TypeScript Library for Creating Mocks. Mockingbird allows you to create class mocks like a breeze with a simple yet powerful @Mock decorator",
  "keywords": [
    "faker",
    "faker-typescript",
    "typescript-faker",
    "typescript-mocks",
    "faker-unit-tests",
    "mockingbird-ts",
    "unit-test",
    "unit-test-mocks",
    "testing-mocks",
    "testing"
  ],
  "contributors": [
    {
      "name": "Omer Morad",
      "email": "omer.moradd@gmail.com"
    },
    {
      "name": "Idan Ptichi",
      "email": "idanpt@gmail.com"
    }
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/omermorad/mockingbird-ts.git"
  },
  "bugs": {
    "url": "https://github.com/omermorad/mockingbird-ts/issues"
  },
  "readme": "https://github.com/omermorad/mockingbird-ts/README.md",
  "scripts": {
    "prebuild": "npx rimraf dist",
    "build": "tsc",
    "watch": "tsc --watch",
    "test": "jest --runInBand --verbose",
    "lint": "eslint '{src,test}/**/*.ts'",
    "lint:fix": "eslint '{src,test}/**/*.ts' --fix"
  },
  "files": [
    "dist",
    "CHANGELOG.md",
    "README.md"
  ],
  "dependencies": {
    "@mockinbird/parser": "^1.0.1-alpha.3",
    "@mockinbird/reflect": "^1.0.1-alpha.3",
    "@mockinbird/types": "^2.0.0-alpha.2"
  },
  "devDependencies": {
    "jest": "27.0.6",
    "rimraf": "^3.0.2",
    "ts-jest": "^27.0.3",
    "ts-loader": "^6.2.2",
    "ts-node": "8.10.2",
    "tsconfig-paths": "^3.9.0",
    "typescript": "^3.9.7"
  },
  "publishConfig": {
    "access": "public"
  }
}
