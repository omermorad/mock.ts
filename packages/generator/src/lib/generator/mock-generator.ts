import { Class } from '@mockinbird/types';
import { ClassParser } from '@mockinbird/parser';
import { MockGeneratorOptions } from '../../types/mock-generator-options.interface';

export class MockGenerator {
  private static readonly DEFAULT_LOCALE = 'en';

  public constructor(private readonly classParser: ClassParser) {}

  /**
   * Return an object with all the properties decorated by the 'Mock' Decorator
   *
   * @example
   * class Person { @Mock() name: string }
   * MockGenerator.create(Person) will return an object { name: <random-string> }
   *
   * @param targetClass
   */
  public create<TClass = any>(targetClass: Class<TClass>): TClass;

  /**
   * Return an array of objects with all the properties decorated by the
   * 'Mock' Decorator (if 'count' is greater than 1)
   *
   * @example
   * class Person { @Mock() name: string }
   * MockGenerator.create(Person, { count: 3, locale: 'es' }) will return an
   * array of objects [{ name: <random-string> }, { name: <random-string> },
   * { name: <random-string> }]
   *
   * Passing a 'locale' property will set a different locale for faker calls
   * The default locale is 'en' (english)
   *
   * @param targetClass
   * @param options
   */
  public create<TClass = any>(targetClass: Class<TClass>, options: MockGeneratorOptions<TClass>): TClass;
  public create<TClass = any>(targetClass: Class<TClass>, options: MockGeneratorOptions<TClass>): TClass[];

  /**
   * Return one or many objects (array) with all the properties decorated
   * by the Mock decorator
   *
   * @param targetClass
   * @param options
   */
  public create<TClass = any>(
    targetClass: Class<TClass>,
    options: MockGeneratorOptions<TClass> = {}
  ): TClass | TClass[] {
    const { count = 1, locale = MockGenerator.DEFAULT_LOCALE, ...config } = options || {};

    this.classParser.setFakerLocale(locale);

    if (count === 1) {
      return this.classParser.parse(targetClass, config);
    }

    const classInstances: TClass[] = [];

    for (let i = 1; i <= count; i++) {
      const parsedClass = this.classParser.parse<TClass>(targetClass, config);
      classInstances.push(parsedClass);
    }

    return classInstances;
  }
}
