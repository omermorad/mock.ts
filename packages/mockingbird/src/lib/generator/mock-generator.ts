import { Class, ClassLiteral } from '@mockinbird/types';
import { ClassParser } from '@mockinbird/parser';
import { MockGeneratorOptions } from '../../types/mock-generator-options.interface';

export class MockGenerator {
  private static readonly DEFAULT_LOCALE = 'en';

  public constructor(private readonly classParser: ClassParser) {}

  private static classToPlain<TClass>(targetClass: TClass): ClassLiteral<TClass> {
    const toPlain = (target: ClassLiteral<TClass>): ClassLiteral<TClass> => {
      for (const key of Object.keys(target)) {
        if (target[key] instanceof Object) {
          target[key] = toPlain({ ...target[key] });
        }
      }

      return target;
    };

    return toPlain({ ...targetClass });
  }

  /**
   * Return an object with all the properties decorated by the 'Mock' Decorator
   *
   * @example
   * class Person { @Mock() name: string }
   * MockGenerator.create(Person) will return an object { name: <random-string> }
   *
   * @param targetClass
   */
  public create<TClass = any>(targetClass: Class<TClass>): TClass;

  /**
   * Return an array of objects with all the properties decorated by the
   * 'Mock' Decorator (if 'count' is greater than 1)
   *
   * @example
   * class Person { @Mock() name: string }
   * MockGenerator.create(Person, { count: 3, locale: 'es' }) will return an
   * array of objects [{ name: <random-string> }, { name: <random-string> },
   * { name: <random-string> }]
   *
   * Passing a 'locale' property will set a different locale for faker calls
   * The default locale is 'en' (english)
   *
   * @param targetClass
   * @param options
   */
  public create<TClass = any>(
    targetClass: Class<TClass>,
    options: MockGeneratorOptions<TClass>
  ): TClass | ClassLiteral<TClass>;

  public create<TClass = any>(
    targetClass: Class<TClass>,
    options: MockGeneratorOptions<TClass>
  ): TClass[] | ClassLiteral<TClass>[];

  /**
   * Return one or many objects (array) with all the properties decorated
   * by the Mock decorator
   *
   * @param targetClass
   * @param options
   */
  public create<TClass = any>(
    targetClass: Class<TClass>,
    options: MockGeneratorOptions<TClass> = {}
  ): TClass | TClass[] | ClassLiteral<TClass> | ClassLiteral<TClass>[] {
    const { count = 1, locale = MockGenerator.DEFAULT_LOCALE, plain = false, ...config } = options || {};

    this.classParser.setFakerLocale(locale);

    if (count === 1) {
      const parsedClass = this.classParser.parse(targetClass, config);
      return plain ? MockGenerator.classToPlain<TClass>(parsedClass) : parsedClass;
    }

    const classInstances: TClass[] = [];
    let parsedClass: TClass | ClassLiteral<TClass>;

    for (let i = 1; i <= count; i++) {
      parsedClass = this.classParser.parse<TClass>(targetClass, config);
      classInstances.push(plain ? MockGenerator.classToPlain<TClass>(parsedClass) : parsedClass);
    }

    return classInstances;
  }
}
