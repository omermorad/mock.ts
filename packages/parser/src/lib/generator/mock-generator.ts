import { Service } from 'typedi';
import { Class, ClassLiteral, isPrimitive } from '@mockingbird/common';
import { MockGeneratorOptions } from '../types/mock-generator-options.interface';
import { ClassParser } from '../parser/class-parser';

@Service()
export class MockGenerator {
  public constructor(private readonly classParser: ClassParser) {}

  private static classToPlain<TClass>(targetClass: TClass): ClassLiteral<TClass> {
    const toPlain = (target: ClassLiteral<TClass>): ClassLiteral<TClass> => {
      for (const key of Object.keys(target)) {
        const value = target[key];

        if (value instanceof Object) {
          if (isPrimitive(value.constructor.name)) {
            toPlain(value);
          } else {
            target[key] = toPlain({ ...value });
          }
        }
      }

      return target;
    };

    return toPlain({ ...targetClass });
  }

  /**
   * Return an object with all the properties decorated by the 'Mock' Decorator
   *
   * @example
   * class Person { @Mock() name: string }
   * MockGenerator.create(Person) will return an object { name: <random-string> }
   *
   * @param targetClass {Class<TClass>}
   * @returns {TClass}
   */
  public generate<TClass = any>(targetClass: Class<TClass>): TClass;

  /**
   * Return an array of objects with all the properties decorated by the
   * 'Mock' Decorator (if 'count' is greater than 1)
   *
   * @example
   * class Person { @Mock() name: string }
   * MockGenerator.create(Person, { count: 3, locale: 'es' }) will return an
   * array of objects [{ name: <random-string> }, { name: <random-string> },
   * { name: <random-string> }]
   *
   * Passing a 'locale' property will set a different locale for faker calls
   * The default locale is 'en' (english)
   *
   * @param targetClass
   * @param options
   */
  public generate<TClass = any>(
    targetClass: Class<TClass>,
    options: MockGeneratorOptions<TClass>
  ): TClass | ClassLiteral<TClass>;

  public generate<TClass = any>(
    targetClass: Class<TClass>,
    options: MockGeneratorOptions<TClass>
  ): TClass[] | ClassLiteral<TClass>[];

  /**
   * Return one or many objects (array) with all the properties decorated
   * by the Mock decorator
   *
   * @param targetClass
   * @param options
   */
  public generate<TClass = any>(
    targetClass: Class<TClass>,
    options: MockGeneratorOptions<TClass> = {}
  ): TClass | TClass[] | ClassLiteral<TClass> | ClassLiteral<TClass>[] {
    const { count = 1, plain = false, locale = 'en', ...config } = options || {};

    this.classParser.setLocale(locale);

    if (count === 1) {
      const parsedClass = this.classParser.parse(targetClass, config);
      return plain ? MockGenerator.classToPlain<TClass>(parsedClass) : parsedClass;
    }

    const classInstances: TClass[] = [];

    for (let i = 1; i <= count; i++) {
      const parsedClass = this.classParser.parse(targetClass, config);
      classInstances.push(plain ? MockGenerator.classToPlain<TClass>(parsedClass) : parsedClass);
    }

    return classInstances;
  }
}
