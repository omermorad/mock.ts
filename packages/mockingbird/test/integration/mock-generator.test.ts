import { classParser as parser } from '@mockinbird/parser';
import { TestClasses } from './common/test-classes';
import { MockGenerator } from '../../src/lib/generator/mock-generator';

import TestClassWithAbsoluteValues = TestClasses.TestClassWithAbsoluteValues;
import TestClassWithNoValues = TestClasses.TestClassWithNoValues;
import TestClassWithCallback = TestClasses.TestClassWithCallback;
import TestClassWithEnum = TestClasses.TestClassWithEnum;
import TestClassWithOtherClass = TestClasses.TestClassWithSingleClass;
import TestClassWithMultiClass = TestClasses.TestClassWithMultiClass;
import TestClassWithRegex = TestClasses.TestClassWithRegex;

describe('MockGenerator - Integration Test', () => {
  let result;

  const generator = new MockGenerator(parser);

  describe('given a decorated class', () => {
    scenario('mock decorator with absolute values', () => {
      beforeAll(() => {
        result = generator.create(TestClassWithAbsoluteValues);
      });

      test('then return the exact same values passed in the options of the mock decorator', () => {
        expect(result).toMatchSnapshot({
          date: expect.any(Date),
        });
      });

      describe('and adding mutation keys', () => {
        beforeAll(() => {
          result = generator.create(TestClassWithAbsoluteValues, { mutations: { name: 'Override Name' } });
        });

        test('then return the same absolute value expect the constant name added from overrides', () => {
          expect(result.name).toBe('Override Name');
        });
      });

      describe('and adding omit keys', () => {
        beforeAll(() => {
          result = generator.create(TestClassWithAbsoluteValues, { omit: ['name'] });
        });

        test('then return the same absolute value expect the constant name added from overrides', () => {
          expect(result.name).toBeUndefined();
        });
      });
    });

    scenario('mock decorator with a callback (faker)', () => {
      beforeAll(() => {
        result = generator.create(TestClassWithCallback);
      });

      test('then return random values from faker', () => {
        expect(result).toMatchObject({
          email: expect.any(String),
          name: expect.any(String),
        });
      });
    });

    scenario('mock decorator with an enum decoratorValue', () => {
      beforeAll(() => {
        result = generator.create(TestClassWithEnum);
      });

      test('then return one random decoratorValue (not key)', () => {
        expect(['foo', 111, 'Bazz1234']).toContain(result.someEnumVal);
      });
    });

    scenario('mock decorator with no/empty values', () => {
      beforeAll(() => {
        result = generator.create(TestClassWithNoValues);
      });

      test('then infer the decoratorValue from the type itself', () => {
        expect(result).toMatchSnapshot({
          name: expect.any(String),
          num: expect.any(Number),
          binary: expect.any(Boolean),
          date: expect.any(Date),
        });
      });
    });

    scenario('mock decorator with a single class', () => {
      beforeAll(() => {
        result = generator.create(TestClassWithOtherClass);
      });

      test('then return an object with the given class', () => {
        expect(Object.keys(result.dog)).toEqual(['name', 'points']);
      });
    });

    describe('mock decorator with a regex', () => {
      beforeAll(() => {
        result = generator.create(TestClassWithRegex);
      });

      test('then return string generated to match the regex', () => {
        expect(result.prop1).toBe('1234');
        expect(result.prop2).toBe('regex');
        expect(result.prop2).toMatch(/^[a-z]{4,5}$/);
      });
    });

    scenario('mock decorator with a multi class', () => {
      beforeAll(() => {
        result = generator.create(TestClassWithMultiClass);
      });

      test("then return contain a property 'dogs' which is array of Dog with length of 'count'", () => {
        expect(result.dogs).toBeInstanceOf(Array);
        expect(result.dogs).toHaveLength(3);
      });

      test('then return array of objects with the given class keys', () => {
        expect(result.dogs).toEqual(
          expect.arrayContaining([expect.objectContaining({ name: expect.any(String), points: expect.any(Number) })])
        );
      });
    });

    scenario("mock decorator with 'count' option", () => {
      beforeAll(() => {
        result = generator.create(TestClassWithAbsoluteValues, { count: 4, locale: 'ja' });
      });

      test("then return array with length of 'count'", () => {
        expect(result).toBeInstanceOf(Array);
        expect(result).toHaveLength(4);
      });
    });
  });
});
