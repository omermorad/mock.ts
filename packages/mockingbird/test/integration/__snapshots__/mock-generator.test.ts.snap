// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MockGenerator - Integration Test given a decorated class Scenario:  mock decorator with absolute values then return the exact same values passed in the options of the mock decorator 1`] = `
Object {
  "binary": true,
  "date": Any<Date>,
  "name": "FooBar",
  "num": 1234,
  "objectLiteral": Object {
    "an": "object",
    "literal": true,
    "thiss": "is",
  },
}
`;

exports[`MockGenerator - Integration Test given a decorated class Scenario:  mock decorator with no/empty values then infer the decoratorValue from the type itself 1`] = `
Object {
  "binary": Any<Boolean>,
  "date": Any<Date>,
  "name": Any<String>,
  "num": Any<Number>,
}
`;

exports[`MockGenerator - Integration Test given a decorated class when using the mock decorator with absolute values then return the exact same values passed in the options of the mock decorator 1`] = `
Object {
  "binary": true,
  "date": Any<Date>,
  "name": "FooBar",
  "num": 1234,
  "objectLiteral": Object {
    "an": "object",
    "literal": true,
    "thiss": "is",
  },
}
`;

exports[`MockGenerator - Integration Test given a decorated class when using the mock decorator with absolute values when using the mock decorator with no/empty values then infer the decoratorValue from the type itself 1`] = `
Object {
  "binary": Any<Boolean>,
  "date": Any<Date>,
  "name": Any<String>,
  "num": Any<Number>,
}
`;

exports[`MockGenerator - Integration Test given a decorated class when using the mock decorator with no/empty values then infer the decoratorValue from the type itself 1`] = `
Object {
  "binary": Any<Boolean>,
  "date": Any<Date>,
  "name": Any<String>,
  "num": Any<Number>,
}
`;
